src/lib/base-command.js

```diff
@@ -47,7 +47,11 @@ class BaseCommand extends Command {
 
     this.setupParams(flags);
 
-    if (!flags.quiet && this.constructor.name !== 'SetupCommand') {
+    if (
+      !flags.quiet &&
+      !flags.json &&
+      this.constructor.name !== 'SetupCommand'
+    ) {
```

Comment:
Does this read more easily? 

```suggestion
    if (!(flags.quiet || flags.json || this.constructor.name === 'SetupCommand')) {
```

On a more general note, I dislike the mentioning of individual implementations (i.e. `SetupCommand`) in base classes. Better would be a method, e.g. `isInteractive()` (or whatever really makes sense) that can be overridden in individual commands.